### DISCUSSION TITLE
Use plain channel send ?

### DISCUSSION THREAD

**Discussion by xiaods** [k3s-io/k3s#4543] [Q&A]
https://github.com/k3s-io/k3s/blob/d93c82f9585ed9e0ff7ebe95b1506f4587fffc1c/pkg/cli/cmds/log_linux.go#L90-L92

use plain channel?
```
//reapChildren calls Wait4 whenever SIGCHLD is received
func reapChildren() {
	sigs := make(chan os.Signal, 1)
	signal.Notify(sigs, syscall.SIGCHLD)
	for {
		<-sigs:   // <--- use plain channel?
		for {
			var wstatus syscall.WaitStatus
			_, err := syscall.Wait4(-1, &wstatus, 0, nil)
			for err == syscall.EINTR {
				_, err = syscall.Wait4(-1, &wstatus, 0, nil)
			}
			if err == nil || err == syscall.ECHILD {
				break
			}
		}
	}
}
```

I want to use plan mode on coding

**Comment by brandond** [comment]
I'm confused by what you're asking. What is "plan mode on coding"?

The code is inspired by other golang reaper implementations, such as
https://github.com/ramr/go-reaper/blob/master/reaper.go

**Reply by xiaods** [reply]
because k3s is my project's upstream. came across an report from deepsource.io. So I want to know how.
https://deepsource.io/gh/xiaods/k8e/run/fc0f4876-72ca-4e39-acd8-ad212ce52f99/go/SCC-S1000

**Comment by xiaods** [comment]
My question is whether this code can be optimized

**Reply by brandond** [reply]
You're welcome to open a PR if you think there's improvements to be made.
